pandas/core/series.py

```diff
@@ -2538,6 +2538,10 @@ def append(self, to_append, ignore_index=False, verify_integrity=False) -> "Seri
             to_concat = [self]
             to_concat.extend(to_append)
         else:
+            if not isinstance(to_append, type(self)):
```

Comment:
I think should allow a Series to be appended to a subclassed Series
```suggestion
            if not isinstance(to_append, Series):
```

also need to move the check into above condition to check the items of a sequence

Reply:
or instead of inside the else, loop over to_concat[1:] outside the else

Reply:
Thanks for the hint @simonjayhawkins.
Could you please suggest the better way to do this between the following 2 ways:
1)
```suggestion
for x in to_concat[1:]:
    if not isinstance(x, Series):
        msg = (
            f"to_append should be a Series or list/tuple of Series, "
            f"got {type(to_append).__name__}"
            f'{(" of " + type(x).__name__) if isinstance(to_append, (list, tuple)) else ""}'
        )
        raise TypeError(msg)
```
2)
```suggestion
if any(not isinstance(x, Series) for x in to_concat[1:]):
    msg = (
        f"to_append should be a Series or list/tuple of Series, "
        f"got {type(to_append).__name__}"
    )
    raise TypeError(msg)
```
Both go after the else. If there's a better way to do this, please do share.
Thanks!

Reply:
Test case **test_concatlike_same_dtypes** is failing in pipeline, which is testing the appending of various data types and expecting to catch and raise a TypeError in Concatenator constructor.
With `if not isinstance(x, Series):` to_append will catch & raise TypeError for all data types. 
Hence **test_concatlike_same_dtypes** is failing.

I think we should consider the following change:
```suggestion
           for x in to_concat[1:]:
               if isinstance(x, pd.DataFrame):
                   raise TypeError
               
```
What are your thoughts on this?

Reply:
IIUC correctly from the issue the behaviour of adding a DataFrame needs to be revisited, so I agree that maybe checking for a DataFrame maybe better than checking for not a Series.

Reply:
I prefer the simple loop of step 1 (fails faster) and the simplicity of message 2.